src/components/transport_manager/src/transport_manager_default.cc

```diff
@@ -68,57 +68,64 @@ int TransportManagerDefault::Init(resumption::LastState& last_state) {
     return E_TM_IS_NOT_INITIALIZED;
   }
 
-#ifdef BLUETOOTH_SUPPORT
-  transport_adapter::TransportAdapterImpl* ta_bluetooth =
-      new transport_adapter::BluetoothTransportAdapter(last_state,
-                                                       get_settings());
+#if defined(BLUETOOTH_SUPPORT)
+  if (!ta_bluetooth_) {
+    ta_bluetooth_ = std::unique_ptr<transport_adapter::TransportAdapter>(
+        new transport_adapter::BluetoothTransportAdapter(last_state,
+                                                         get_settings()));
+  }
 #ifdef TELEMETRY_MONITOR
   if (metric_observer_) {
-    ta_bluetooth->SetTelemetryObserver(metric_observer_);
+    ta_bluetooth_->SetTelemetryObserver(metric_observer_);
   }
 #endif  // TELEMETRY_MONITOR
-  AddTransportAdapter(ta_bluetooth);
-  ta_bluetooth = NULL;
-#endif
-
-  const uint16_t port = get_settings().transport_manager_tcp_adapter_port();
-  transport_adapter::TransportAdapterImpl* ta_tcp =
-      new transport_adapter::TcpTransportAdapter(
-          port, last_state, get_settings());
+  AddTransportAdapter(ta_bluetooth_.release());
```

Comment:
I can see 2 issues with such approach :

1. It is not obvious that created ptr life cycle is longer then unique_ptr. release method is not widely used. If you wan't to take care about life cycle of transport adapters, please move unique ptr to the component that is responsible for TA life time. 

2. Can't see reason why pointers moved to class members. With such approach, after `Init` method class members will contain non initialized unique ptrs.

I would propose to use some kind of custom AutoGuard type to guarantee that pointers will be deleted if the was not sent to `AddTransportAdapter`. But avoid using unique_ptr because it a little bit confusing. 

Example : 
```
class AutoExecutor  { // Create better name
  AutoExecutor(std::function<void ()>& function) function_(function){}
  ~AutoExecutor(){function_();}
  std::function<void ()> function_;
};
ta_bluetooth = new ...
AutoExecutor bt_deleter ([&ta_bluetooth_]{delete ta_bluetooth_});
// [Code space 1] ... 
AddTransportAdapter(ta_bluetooth);
ta_bluetooth = nullptr;
```

 * If  `[Code space 1]` will throw an error, ta_bluetooth will be deleted by bt_deleter;
 * In regular execution flow bt_deleter will delete `nullptr`, that is safe.

Reply:
@LuxoftAKutsan 
Thanks, fixed in:https://github.com/smartdevicelink/sdl_core/pull/3134/commits/e8a1c8ed50ecbc4689536c888edca85d38943cd4